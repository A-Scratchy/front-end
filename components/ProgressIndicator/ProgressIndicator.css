.ProgressIndicator {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 25px 0;
}

.bar {
  width: 50%;
  margin-top: 10px;
  border-radius: 8px;
  background-color: #bfc9d9;
  height: 10px;
}

.progress {
  border-radius: 8px;
  background-color: #252e3e;
  height: 10px;
}
