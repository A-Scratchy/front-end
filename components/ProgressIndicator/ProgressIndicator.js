import React from 'react';
import { number } from 'prop-types';
import styles from './ProgressIndicator.css';

const ProgressIndicator = ({ stepNumber, totalSteps }) => {
  if (totalSteps < 1) return null;

  const percentageCompleted = (stepNumber / totalSteps) * 100;

  return (
    <div className={styles.ProgressIndicator}>
      <div>
        {stepNumber}/{totalSteps} Complete
      </div>
      <div className={styles.bar}>
        <div className={styles.progress} style={{ width: `${percentageCompleted}%` }} />
      </div>
    </div>
  );
};

ProgressIndicator.propTypes = {
  stepNumber: number,
  totalSteps: number,
};

ProgressIndicator.defaultProps = {
  stepNumber: 0,
  totalSteps: 0,
};

export default ProgressIndicator;
