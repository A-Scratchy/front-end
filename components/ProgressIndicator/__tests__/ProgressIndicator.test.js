import React from 'react';
import createSnapshotTest from 'test-utils/createSnapshotTest';

import ProgressIndicator from '../ProgressIndicator';

describe('ProgressIndicator', () => {
  it('should render with required props', () => {
    createSnapshotTest(<ProgressIndicator>Test</ProgressIndicator>);
  });

  it('should render with many props assigned', () => {
    createSnapshotTest(<ProgressIndicator className="test-class">Test</ProgressIndicator>);
  });
});
