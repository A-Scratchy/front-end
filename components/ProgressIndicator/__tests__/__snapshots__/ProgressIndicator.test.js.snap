// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressIndicator should render with many props assigned 1`] = `null`;

exports[`ProgressIndicator should render with required props 1`] = `null`;
